examples/sod/ArborX_SOD.hpp

```diff
@@ -0,0 +1,383 @@
+          Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                             "ArborX::SOD::critical_bin_values"),
+          num_critical_bin_particles);
+      {
+        auto offsets = clone(critical_bin_offsets);
+        _bvh.query(exec_space, SOD::ParticlesWrapper<Particles>{_particles},
+                   Details::CriticalBinParticles<MemorySpace, Particles>{
+                       _particles, critical_bin_ids, _fof_halo_centers,
+                       num_sod_bins, _r_min, _r_max, offsets,
+                       critical_bin_values},
+                   Experimental::TraversalPolicy().setPredicateSorting(
+                       sort_predicates));
+      }
+
+      Kokkos::Profiling::popRegion();
+
+      // Sort the particles based on their distance to the corresponding FOF
+      // center
+      Kokkos::Profiling::pushRegion(
+          "ArborX::SODHandle::computeRdelta::sort_values");
+      sortObjects(exec_space, critical_bin_values);
+      Kokkos::Profiling::popRegion();
+
+      // Compute R_delta
+      Kokkos::Profiling::pushRegion(
+          "ArborX::SODHandle::compute_R_delta::compute_r_delta");
+      Kokkos::resize(sod_halo_rdeltas_index, 0);
+      std::tie(sod_halo_rdeltas, sod_halo_rdeltas_index) =
+          Details::computeSODRdeltas(
+              exec_space, rho, rho_ratio, particle_masses, sod_halo_bin_masses,
+              critical_bin_ids, critical_bin_offsets, critical_bin_values);
+      Kokkos::Profiling::popRegion();
+
+      Kokkos::parallel_for(
+          "ArborX::SODHandle::compute_R_delta::update_rdeltas_index",
+          Kokkos::RangePolicy<ExecutionSpace>(0, num_halos),
+          KOKKOS_LAMBDA(int const halo_index) {
+            for (int bin_id = 0; bin_id < critical_bin_ids(halo_index);
+                 ++bin_id)
+              sod_halo_rdeltas_index(halo_index) +=
+                  sod_halo_bin_counts(halo_index, bin_id);
+          });
+
+      Kokkos::Profiling::popRegion();
+    }
+    else
+    {
+      Kokkos::Profiling::pushRegion("ArborX::SODHandle::compute_R_delta_no_bins");
+
+      Kokkos::View<int *, MemorySpace> offsets;
+      Kokkos::View<int *, MemorySpace> indices;
+      query(exec_space, offsets, indices);
+
+      using ParticlesAccess = ArborX::AccessTraits<Particles, PrimitivesTag>;
+
+      using TeamPolicy =
+          Kokkos::TeamPolicy<ExecutionSpace, Kokkos::Schedule<Kokkos::Dynamic>>;
+      using team_member = typename TeamPolicy::member_type;
+
+      ArborX::reallocWithoutInitializing(sod_halo_rdeltas, num_halos);
+      ArborX::reallocWithoutInitializing(sod_halo_rdeltas_index, num_halos);
+      Kokkos::deep_copy(sod_halo_rdeltas_index, INT_MAX);
+
+      // Avoid capturing *this
+      auto particles = _particles;
+      auto fof_halo_centers = _fof_halo_centers;
+
+      Kokkos::parallel_for(
+          "ArborX::SODHandle::computeRdelta::compute_rdelta_index",
+          TeamPolicy(exec_space, num_halos, 512),
```

Comment:
Not actually sure it's a good idea to hardcode the team size here. Should probably switch to `Kokkos::AUTO`.